packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -240,16 +230,13 @@ async function redeemSharingLink(
 			let redeemUrl: string | undefined;
 			async function callSharesAPI(baseUrl: string): Promise<void> {
 				await getWithRetryForTokenRefresh(async (tokenFetchOptions) => {
-					const storageToken = await storageTokenFetcher(
-						tokenFetchOptions,
-						"RedeemShareLink",
-					);
 					redeemUrl = `${baseUrl}/_api/v2.0/shares/${encodedShareUrl}`;
-					const { url, headers } = getUrlAndHeadersWithAuth(
-						redeemUrl,
-						storageToken,
-						forceAccessTokenViaAuthorizationHeader,
+					const url = redeemUrl;
+					const storageToken = await getAuthHeader(
+						{ ...tokenFetchOptions, request: { url, method: "GET" } },
```

Comment:
(nit) Should we be passing method down (to fetchAndParseAsJSONHelper)? Maybe it has no other way to make such call (today), but if it changes and it does not use GET, this flow will get broken.